app/revenue/migrations/0003_auto_20190123_2218.py

```diff
@@ -0,0 +1,102 @@
+# Generated by Django 2.1.2 on 2019-01-23 22:18
+
+import datetime
+from django.db import migrations, models
+import django.db.models.deletion
+import economy.models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('grants', '0006_grant_request_ownership_change'),
+        ('revenue', '0002_auto_20190123_2203'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Coupon',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('created_on', models.DateTimeField(db_index=True, default=economy.models.get_time)),
+                ('modified_on', models.DateTimeField(default=economy.models.get_time)),
+                ('code', models.CharField(help_text='The Coupon Code', max_length=255)),
+                ('discount_per_period', models.DecimalField(decimal_places=2, help_text='Discount, per period, to be applied to this plan in USD. Must not be more than the price per period of the plan.', max_digits=50)),
+                ('start_date', models.DateTimeField(default=datetime.datetime(1990, 1, 1, 0, 0), help_text='The start date for validity of this coupon')),
+                ('end_date', models.DateTimeField(default=datetime.datetime(1990, 1, 1, 0, 0), help_text='The end date for validity of this coupon')),
+            ],
+            options={
+                'abstract': False,
+            },
+        ),
+        migrations.CreateModel(
+            name='Plan',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('created_on', models.DateTimeField(db_index=True, default=economy.models.get_time)),
+                ('modified_on', models.DateTimeField(default=economy.models.get_time)),
+                ('slug', models.CharField(help_text='The Slug', max_length=255)),
+                ('name', models.CharField(help_text='The SKU Name', max_length=255)),
+                ('cost_per_period', models.DecimalField(decimal_places=2, help_text='Cost of this plan in USD', max_digits=50)),
+                ('period_length_seconds', models.PositiveIntegerField()),
```

Comment:
Looking at the spec and planning docs for this feature, I'm wondering if this will lead to confusion among users when the plan needs to be recharged every 30 days instead of on the same day each month. Looks like a 30-day month is used to calculate the period. A slightly more complicated path forward could use a `DurationField` to store a `timedelta` of 1 month and use that for plan expiry calculations. A simple path forward using the current code would be to change the pricing language from per month to per 30 days.